ssm/crypto.py

```diff
@@ -64,23 +64,36 @@ def check_cert_key(certpath, keypath):
     if cert == key:
         return False
 
-    p1 = Popen(['openssl', 'x509', '-pubkey', '-noout'],
-               stdin=PIPE, stdout=PIPE, stderr=PIPE, universal_newlines=True)
-    pubkey1, error = p1.communicate(cert)
+    try:
+        certificate = OpenSSL.crypto.load_certificate(
+            OpenSSL.crypto.FILETYPE_PEM, cert
+        )
+        crypto_public_key = certificate.get_pubkey()
+        public_key_bytes = OpenSSL.crypto.dump_publickey(
+            OpenSSL.crypto.FILETYPE_PEM, crypto_public_key
+        )
 
-    if error != '':
+        certificate_public_key = public_key_bytes.decode("utf-8")
```

Comment:
This seems like an unnecessary step if we're doing this to both keys.

Reply:
resolved in next commit